pkg/util/spanlogger/spanlogger.go

```diff
@@ -94,3 +95,12 @@ func (s *SpanLogger) Error(err error) error {
 	s.Span.LogFields(otlog.Error(err))
 	return err
 }
+
+// triggerError is used to detect error in kvps, and calls s.Error if there's any.
+func (s *SpanLogger) triggerError(kvps ...interface{}) {
```

Comment:
This would set the error flag for any error logged, regardless of the log level. I think the idea in #3016 was to detect the log level and set the error flag only if the `level` was `error`. We have may have cases where we log an error even if it's not critical and way may not want to "red flag" it in the span, so filtering out any non error level may help.